### ISSUE TITLE
508-defect-2 [SCREENREADER]: Search field `form` elements should have the landmark role="search"

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#26360]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I expect search fields on the page to be discoverable using my landmarks list.

## Details

> Landmarks can be used by assistive technology to quickly identify and navigate to large sections of the document. The search role is added to the container element that encompasses the items and objects that, as a whole, combine to create search functionality. When a <form> is a search form, use the search role instead of form. If a document includes more than one search, each should have a unique label. While there is an input of type search, there is no HTML element that defines a search landmark.

[MDN web docs on the search role.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Search_role)

All search field forms on the page should have the attribute `role="search"` to help screen reader users easily find or track back to the search field using their landmarks or regions list.

For example:
``` html
<form role="search">
  <!-- search input -->
</form>
```

The mobile resources search results page is the one instance where this _isn't_ possible, which would break consistency for mobile screen reader users.

<img width="1435" alt="Screen Shot 2021-06-21 at 3 07 49 PM" src="https://user-images.githubusercontent.com/14154792/122814654-81413000-d2a2-11eb-9771-0d1ec12548bb.png">

Because this is a `button` that _contains_ the search form, we can't override its role into `search`. That means that the search _won't_ be discoverable to mobile screen reader users when navigating by landmarks or regions.

## Acceptance Criteria

- [ ] All search `form`s have `role="search"`

**Comment by denisecoveyduc** [comment]
@brianalloyd  UI Consistency feedback we have received for Resources and Support. Passing it to PW but please let me know if we can help in any way.

**Comment by joshkimux** [comment]
@denisecoveyduc I should have clarified on the ticket, but this is global to all search fields across find a form, resources and support, and va.gov 🙇

**Comment by denisecoveyduc** [comment]
Hi @joshkimux Thanks. I neglected to mention that @srsuddath has already resolved for va.gov global search.

**Comment by joshkimux** [comment]
@denisecoveyduc Great! @srsuddath by any chance is this already on staging for me to validate? 😄

**Comment by Rleahy22** [comment]
For validation: I used [this extension](http://matatk.agrip.org.uk/landmarks/) to view landmarks but other a11y tools will work as well. The search form should show up in landmarks now

**Comment by joshkimux** [comment]
@Rleahy22 that's a great extension! Thanks for sharing.
Quick question, would it be possible to add an `aria-labelledby` that points to the `h1` to that div? It will provide a more unique label to the landmark.

<img width="1792" alt="Screen Shot 2021-07-21 at 3 13 54 PM" src="https://user-images.githubusercontent.com/14154792/126546563-5db1a211-1ce4-4caf-88ed-69bfd49766b6.png">

**Comment by Rleahy22** [comment]
@joshkimux I added the `aria-labelledby` via https://github.com/department-of-veterans-affairs/vets-website/pull/17958

**Comment by joshkimux** [comment]
Thank you @Rleahy22 ! Almost there. Some last changes:
- `role="search"` and `aria-labelledby="unique-id"` should be attributes to the `div` that is the parent to `Enter a keyword`
- `role="search"` and `aria-labelledby="unique-id"` should be removed from the `div` that is a child to the `form`

This will make sure that only one search landmark exists on the page. If we have two `role="search"` it will render as two separate search landmarks, which is undesired. Other than that, `aria-labelledby` is pointing successfully to the `h1`, so it's just a matter of relocating that attribute to the parent `div` 😄 

<img width="1791" alt="Screen Shot 2021-07-22 at 10 09 31 AM" src="https://user-images.githubusercontent.com/14154792/126653400-cae5db1f-ff58-4a72-9246-687c67ca431f.png">

**Comment by Rleahy22** [comment]
@joshkimux fixed!

**Comment by joshkimux** [comment]
Wonderful! I'll detach this from the VA.gov search results staging review and assign it to the other search teams to follow! 🥳 
<img width="1402" alt="Screen Shot 2021-07-22 at 4 03 41 PM" src="https://user-images.githubusercontent.com/14154792/126702547-46ce749f-0c2a-4a53-b071-1c6de8b44212.png">

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO and @zacharymorel , would the work @Rleahy22 did be replicable for resources and find a form?

**Comment by MarciMcGuireGCIO** [comment]
^ Looping in @brianalloyd and @kelsonic for Resources & Support.

**Comment by brianalloyd** [comment]
@joshkimux we've resolved the PW affected areas, please review on your end in support of close out on this ticket.  Thanks.

**Comment by joshkimux** [comment]
<img width="1524" alt="Screen Shot 2021-07-27 at 12 37 17 PM" src="https://user-images.githubusercontent.com/14154792/127193192-de905810-0ca0-4217-870b-da88dba279d1.png">

Confirmed for resources and support! Great work, I'll unassign y'all from the ticket @brianalloyd

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux we'll plan this so it can go in with our next update on Find a VA Form.  Attaching to that Epic. #26816

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel this one looks pretty straightforward.  It flew under the radar because it wasn't attached to our old 508 epic.

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel is this something that can be included in the modal work?

**Comment by laflannery** [comment]
Global, Find a Form and Resources search all have the proper landmark.  Will close in 1 week unless told otherwise
